winml/test/api/LearningModelSessionAPITest.cpp

```diff
@@ -475,7 +475,135 @@ static void WindowFunction(const wchar_t* window_operator_name, TensorKind kind)
+          .SetInput(L"input", L"IDFT.Output.1")
+          .SetAttribute(L"axis", TensorInt64Bit::CreateFromArray({}, {INT64(0)}))
+          .SetOutput(L"output", L"IDFT.Output.2"))
+        .Operators().Add(Operator(L"ReduceSumSquare")
+          .SetInput(L"data", L"DFT.Output.2")
+          .SetAttribute(L"axes", TensorInt64Bit::CreateFromArray({1}, {3}))
+          .SetAttribute(L"keepdims", TensorInt64Bit::CreateFromArray({}, {0}))
+          .SetOutput(L"reduced", L"magnitude_squared"))
+        .Operators().Add(Operator(L"Sqrt")
+          .SetInput(L"X", L"magnitude_squared")
+          .SetOutput(L"Y", L"sqrt_magnitude"))
+        .Operators().Add(Operator(L"ReduceSumSquare")
+          .SetInput(L"data", L"IDFT.Output.2")
+          .SetAttribute(L"axes", TensorInt64Bit::CreateFromArray({1}, {3}))
+          .SetAttribute(L"keepdims", TensorInt64Bit::CreateFromArray({}, {0}))
+          .SetOutput(L"reduced", L"magnitude_squared2"))
+        .Operators().Add(Operator(L"Sqrt")
+          .SetInput(L"X", L"magnitude_squared2")
+          .SetOutput(L"Y", L"sqrt_magnitude2"))
+        .Operators()
+          .Add(Operator(L"Reshape")
+          .SetInput(L"data", L"sqrt_magnitude")
+          .SetConstant(L"shape", TensorInt64Bit::CreateFromArray({4}, {INT64(1), INT64(1), INT64(height), INT64(width) }))
+          .SetOutput(L"reshaped", L"Output.Spectra"))
+        .Operators()
+          .Add(Operator(L"Reshape")
+          .SetInput(L"data", L"sqrt_magnitude2")
+          .SetConstant(L"shape", TensorInt64Bit::CreateFromArray({4}, {INT64(1), INT64(1), INT64(height), INT64(width) }))
+          .SetOutput(L"reshaped", L"Output.Inverse"))
+        .Operators().Add(Operator(L"Sub")
+          .SetInput(L"A", L"Input.Signal")
+          .SetInput(L"B", L"Output.Inverse")
+          .SetOutput(L"C", L"Output.Error"));
+
+  auto model = builder.CreateModel();
+  
+  LearningModelSession session(model);
+  LearningModelBinding binding(session);
+
+  // Bind input
+  binding.Bind(L"Input.Signal", frame);
+
+  // Bind output
+  auto spectra = VideoFrame(BitmapPixelFormat::Bgra8, INT32(width), INT32(height));
+  binding.Bind(L"Output.Spectra", spectra);
+  auto inverse = VideoFrame(BitmapPixelFormat::Bgra8, INT32(width), INT32(height));
+  binding.Bind(L"Output.Inverse", inverse);
+
+  // Evaluate
+  auto start = std::chrono::high_resolution_clock::now();
+  auto result = session.Evaluate(binding, L"");
+  auto end = std::chrono::high_resolution_clock::now();
+  std::chrono::duration<double, std::micro> evaluate_duration_in_microseconds = end - start;
+  printf("Evaluate Took: %fus\n", evaluate_duration_in_microseconds.count());
+
+  auto error = result.Outputs().Lookup(L"Output.Error").as<TensorFloat>();
+  auto error_ivv = error.GetAsVectorView();
+  for (auto i = 0; i < height * width; i++) {
+    WINML_EXPECT_TRUE(abs(error_ivv.GetAt(i)) < .001);
+  }
+
+  SaveSoftwareBitmap(L"fft2d.jpg", spectra.SoftwareBitmap());
```

Comment:
what do you do with these files? I don't see them referenced anywhere else.

Reply:
I look at them to make sure they look correct ;)